Platform/UTMData.swift

```diff
@@ -468,6 +468,113 @@ class UTMData: ObservableObject {
         }
     }
     
+    // MARK: - External Drive Disk Image Functions
+
+    func importDriveFromExternal(_ image: URL, for config: UTMConfiguration, interface: String? = nil) throws {
+        _ = image.startAccessingSecurityScopedResource()
+        defer { image.stopAccessingSecurityScopedResource() }
+
+        let path = image.appendingPathExtension("bookmark").lastPathComponent
+
+        var isDir: ObjCBool = false
+        guard fileManager.fileExists(atPath: image.path, isDirectory: &isDir), !isDir.boolValue else {
+            if image.pathExtension == "utm" {
+                throw NSLocalizedString("You cannot import a .utm package as a disk image. Did you mean to open the package with UTM?", comment: "UTMData")
```

Comment:
Why are you throwing Strings? If you need this to be a public function I would expect it to use the `self.alertMessage` like you did further down. Otherwise, please consider changing this and similar functions to `private`.

Reply:
If you look elsewhere in UTMData the code also throws Strings because calls to UTMData functions are done in a `data.busyWork` block which catches exceptions and sets the alert message. The manually setting alertmessage was only inside the SavePanel closure because you can't throw from there into the caller.